src/Function/Definitions.agda

```diff
@@ -0,0 +1,40 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Definitions for types of functions
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+open import Relation.Binary
+
+module Function.Definitions
+  {a b ℓ₁ ℓ₂} {A : Set a} {B : Set b} (_≈₁_ : Rel A ℓ₁) (_≈₂_ : Rel B ℓ₂)
+  where
+
+open import Data.Product using (∃; _×_)
+open import Level using (_⊔_)
+
+------------------------------------------------------------------------
+-- Definitions
+
+Congruent : (A → B) → Set (a ⊔ ℓ₁ ⊔ ℓ₂)
+Congruent f = ∀ {x y} → x ≈₁ y → f x ≈₂ f y
+
+Injective : (A → B) → Set (a ⊔ ℓ₁ ⊔ ℓ₂)
+Injective f = ∀ {x y} → f x ≈₂ f y → x ≈₁ y
+
+Surjective : (A → B) → Set (a ⊔ b ⊔ ℓ₂)
+Surjective f = ∀ x → ∃ λ y → y ≈₂ f x
+
+Bijective : (A → B) → Set (a ⊔ b ⊔ ℓ₁ ⊔ ℓ₂)
+Bijective f = Injective f × Surjective f
+
+LeftInverses : (A → B) → (B → A) → Set (b ⊔ ℓ₂)
```

Comment:
I've added an extra `s` here because at the both the package and the property would be called `LeftInverse`. I'd definitely welcome better suggestions.

Reply:
Already taken by the structure :cry:  so we have `LeftInverse` for the package, `IsLeftInverse` for the structure and then we need a third name for the property...

Reply:
Make it infix and call it `_IsLeftInverseOf_`?

Reply:
That is one solution, but then it no longer works if the equality module parameters are not provided. Definitely better then at the moment though!

Reply:
What about `Inverseˡ`, `Inverseʳ`, and `Inverseᵇ` then?
We have now multiple places already where `ᵇ` means both left & right.

Reply:
Yup, excellent suggestion, thanks!